File: leet-code/largestPerimeter.go
package main

import "sort"

func largestPerimeter(A []int) int {
	sort.Slice(A, func(a, b int) bool {
		return A[a] > A[b]
	})

	for i := 0; i < len(A)-2; i++ {
		if A[i] < A[i+1]+A[i+2] {
			return A[i] + A[i+1] + A[i+2]
		}
	}
	return 0
}

File: leet-code/largestPerimeter.js
/**
 * @param {number[]} A
 * @return {number}
 */
var largestPerimeter = function (A) {
  if (A.length < 3) {
    return 0
  }
  A = A.sort((a, b) => b - a)

  for (let i = 0, len = A.length; i < len - 2; i++) {
    if (A[i] < A[i + 1] + A[i + 2]) {
      return A[i] + A[i + 1] + A[i + 2]
    }
  }
  return 0
}

console.log(largestPerimeter([1, 2, 1]))
